src/plugins/usage_collection/server/collector/collector_set.test.ts

```diff
@@ -698,6 +714,32 @@ describe('CollectorSet', () => {
       expect(results).toHaveLength(2);
     });
 
+    it('calls fetch with execution context', async () => {
+      collectorSet.registerCollector(
+        collectorSet.makeUsageCollector({
+          type: 'ready_col',
+          isReady: () => true,
+          schema: { test: { type: 'long' } },
+          fetch: () => ({ test: 1000 }),
+        })
+      );
+
+      const mockEsClient = elasticsearchServiceMock.createClusterClient().asInternalUser;
+      const mockSoClient = savedObjectsClientMock.create();
+      await collectorSet.bulkFetch(mockEsClient, mockSoClient, undefined);
+
+      expect(executionContext.withContext).toHaveBeenCalledTimes(1);
+      expect(executionContext.withContext).toHaveBeenCalledWith(
+        {
+          type: 'usage_collection',
+          name: 'collector.fetch',
```

Comment:
NIT/Optional: Can we adapt this test (or add a new one) with more than one collector just to make sure that `executionContext.withContext` is called with the correct parameter for each of them?

Reply:
Done!